plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObjectPdo.class.php

```diff
@@ -1276,7 +1284,20 @@ public function serialize()
     foreach ($this->getNameAccessPoints() as $item)
     {
       $node = new arElasticSearchActorPdo($item->id);
-      $serialized['names'][] = $node->serialize();
+
+      $names = array(
+        'id' => $node->id,
+        'i18n' => arElasticSearchModelBase::serializeI18ns(
+          $node->id,
+          array('QubitActor'),
+          array('fields' => array('authorized_form_of_name'))
+        )
+      );
+
+      // Add other names, parallel names, and standardized names
+      $names += $node->serializeAltNames();
```

Comment:
This will probably not work if you remove the names from the full foreign types as the other names mapping will be missing in the new names mapping.

Reply:
I've tested with a linked authoirty record with an "other names" value, and it is stored with the "names" data structure.

e.g.
```
"names": [
            {
              "id": "79776",
              "i18n": {
                "en": {
                  "authorizedFormOfName": "Roy, Flora"
                },
                "languages": [
                  "en"
                ]
              },
              "otherNames": [
                {
                  "sourceCulture": "en",
                  "i18n": {
                    "en": {
                      "name": "Floopy"
                    },
                    "languages": [
                      "en"
                    ]
                  }
                }
              ]
            }
          ],
```

Reply:
I wonder when is the `otherNames` mapping inside `names` defined not the the full foreign type is not in there, since the partial foreign type has `dynamic: strict` it should not be possible to add it if it's not defined.

Have you fully recreated the index after the latest changes? Please, could you verify in the mapping definition the final `names` mapping. Just the mapping, as requesting a single IO will only show the populated data in the document.